Userland/Utilities/comm.cpp

```diff
@@ -58,31 +61,38 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
         return 1;
     }
 
-    auto process_file = [](String const& path, auto& file, int file_number) {
-        if (path == "-") {
-            file = Core::File::standard_input();
-        } else {
-            auto file_or_error = Core::File::open(path, Core::OpenMode::ReadOnly);
-            if (file_or_error.is_error()) {
-                warnln("Failed to open file{} '{}': {}", file_number, path, file_or_error.error());
-                return false;
-            }
-            if (Core::File::is_directory(path)) {
-                warnln("Failed to open file{} '{}': is a directory", file_number, path);
-                return false;
-            }
-            file = file_or_error.value();
+    auto open_file = [](String const& path, auto& file, int file_number) {
```

Comment:
Is this function really needed?

Reply:
Without it, the logic would be duplicated. Line 88 calls this twice, once for each file.

Reply:
Yep, but why are you making a function which handles errors manually instead of using `TRY(...)`? Also I'm unsure why you check if the path is a directory *after* opening it.